### PR TITLE
WB-1588: Allow custom ActionItem elements

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

1. Allow custom `ActionItem` components by using `Cell` internally.

- Removed `ClickableBehavior` from `ActionItem` and replaced it with
  `CompactCell` (which internally uses `Clickable`).
- Removed `skipClientNav` from `ActionItem` as it is no longer used/needed.
- Added stories and better documentation for `ActionMenu` and `ActionItem`.

2. Modified Cell to support two new props required for the `ActionItem` changes:
- `role` is used to set the `role` attribute on the cell's root element.
- `rootStyle` is used to override the `style` attribute on the cell's root
  element.

Issue: WB-1588

### ActionMenu:

1. Navigate to http://localhost:6061/?path=/story/dropdown-actionmenu--custom-action-items
2. Verify that the custom action items are rendered as expected.
3. Verify that the custom action items are clickable and that the click handler
   is called as expected.

https://github.com/Khan/wonder-blocks/assets/843075/b4ba22c8-5472-466c-a72a-045c27e22942

### ActionItem docs:

1. Navigate to http://localhost:6061/?path=/docs/dropdown-actionitem--docs
2. Verify that the documentation for `ActionItem` is correct and up to date.

Screenshot 2023-12-05 at 12 52 01 PM

### CODE REVIEW

File: __docs__/wonder-blocks-dropdown/action-item.stories.tsx
Comment (jandrade) [inline __docs__/wonder-blocks-dropdown/action-item.stories.tsx:1]: **note:** Added separate docs for `ActionItem` to help understanding better how this component works.

File: __docs__/wonder-blocks-dropdown/action-menu.stories.tsx [hunk 1]
```diff
@@ -431,3 +442,91 @@ const locales = [
     {id: "fr", locale: "fr", localName: "français"},
     {id: "it", locale: "it", localName: "italiano"},
 ];
+
```
Comment (jandrade) [inline __docs__/wonder-blocks-dropdown/action-menu.stories.tsx:445]: **note:** This story shows how we can use an `ActionMenu` with custom `ActionItem` elements.

File: consistency-tests/__tests__/clickables.test.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ import {render, screen} from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 
 import plus from "@phosphor-icons/core/regular/plus.svg";
-import {ActionItem, OptionItem} from "@khanacademy/wonder-blocks-dropdown";
+import {OptionItem} from "@khanacademy/wonder-blocks-dropdown";
```
Comment (jandrade) [inline consistency-tests/__tests__/clickables.test.tsx:13]: **note:** Removed `ActionItem` from here as we now delegate the ActionItem's clickable behavior/rendering to `CompactCell`.

File: packages/wonder-blocks-cell/src/components/internal/cell-core.tsx [hunk 3]
```diff
@@ -216,8 +221,10 @@ const styles = StyleSheet.create({
         background: Color.white,
         color: Color.offBlack,
         display: "flex",
-        minHeight: CellMeasurements.cellMinHeight,
+        // minHeight: CellMeasurements.cellMinHeight,
+        minHeight: 40,
```
Comment (jandrade) [inline packages/wonder-blocks-cell/src/components/internal/cell-core.tsx:225]: TODO(juan): Change this by passing that value via `rootStyle`.

File: packages/wonder-blocks-dropdown/src/components/action-item.tsx [hunk 4]
```diff
@@ -115,131 +115,101 @@ export default class ActionItem extends React.Component<ActionProps> {
...
+            />
         );
     }
 }
 
 const styles = StyleSheet.create({
+    wrapper: {
+        // This removes the 300ms click delay on mobile browsers by indicating
+        // that "double-tap to zoom" shouldn't be used on this element.
+        touchAction: "manipulation",
```
Comment (jeresig) [inline packages/wonder-blocks-dropdown/src/components/action-item.tsx:183]: Ooh - neat!

File: packages/wonder-blocks-dropdown/src/components/__tests__/action-item.test.tsx [hunk 5]
```diff
@@ -18,7 +18,10 @@ describe("ActionItem", () => {
         render(<ActionItem href="/foo" label="Example" disabled={true} />);
 
         // Assert
-        expect(screen.getByRole("menuitem")).toBeDisabled();
+        expect(screen.getByRole("menuitem")).toHaveAttribute(
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/__tests__/action-item.test.tsx:21]: **note:** Cell handles the disabled state with the `aria-disabled` attr.

**Review by jeresig — APPROVED** [review]
This looks really sharp - thank you for working on this!

File: packages/wonder-blocks-dropdown/src/components/action-item.tsx [hunk 6]
```diff
@@ -41,22 +37,9 @@ type ActionProps = {
...
-     * If the URL passed to href is local to the client-side, e.g.
-     * /math/algebra/eval-exprs, then it tries to use react-router-dom's Link
-     * component which handles the client-side navigation. You can set
-     * `skipClientNav` to true avoid using client-side nav entirely.
-     *
-     * NOTE: All URLs containing a protocol are considered external, e.g.
-     * https://khanacademy.org/math/algebra/eval-exprs will trigger a full
-     * page reload.
-     */
-    skipClientNav?: boolean;
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/action-item.tsx:59]: **note:** This is no longer used, so I got rid of it.

**Pull request merged**